internal/engine/client.go

```diff
@@ -104,6 +96,28 @@ func waitBuildkit(ctx context.Context, host string) ([]*bkclient.WorkerInfo, err
 		retryAttempts = 1800
 	)
 
+	var c *bkclient.Client
+	var err error
+
+	for retry := 0; retry < retryAttempts; retry++ {
+		c, err = bkclient.New(ctx, host)
```

Comment:
I just remembered that GRPC has confusing behavior around ignoring certain errors and using its own retrying in the background, so I think for this call we should probably include the `bkclient.WithFailFast` option to be safe.

Also, given this sort of hidden complication, I think it's actually pretty important to add a test case like what we talked about in discord.